### ISSUE TITLE
Not compatable with ngrok

### ISSUE THREAD

**Issue by ElvizLai** [mholt/caddy#182]
![qq20150710-1](https://cloud.githubusercontent.com/assets/501861/8610388/04faa28a-26e4-11e5-8d5e-23314f8ae663.png)

![qq20150710-2](https://cloud.githubusercontent.com/assets/501861/8610448/b724627a-26e4-11e5-8c1a-2bb8c834bf8f.png)

I'm not quite sure this is a caddy bug. So I tried to build a server client by myself with go, the client with ngrok works well.

**Comment by mholt** [comment]
Huh. That's odd. Thanks for the helpful screenshots!

What if your Caddyfile used :8000 instead of 127.0.0.1:8000? (That should allow requests to any host to connect to it.)

Disclaimer: I've never used ngrok more than just experimentally, so I'm going off intuition here.

**Comment by ElvizLai** [comment]
@mholt Changing to :8000 works, but this config also expose my port 8000 to public.

**Comment by mholt** [comment]
Yeah, I wouldn't expect you to have to do that, but it's good to know that works. Will look into the problem more once I'm done traveling. Anyone else that has ideas are welcome to chime in.

**Comment by mholt** [comment]
Oh. Can you find out what the Host field of the header is that is being sent to Caddy from ngrok? Is it 127.0.0.1 or elvizlai....xyz?

**Comment by ElvizLai** [comment]
@mholt 

![qq20150710-4](https://cloud.githubusercontent.com/assets/501861/8614826/335088b6-2720-11e5-8658-02e172b113a5.png)

![qq20150710-5](https://cloud.githubusercontent.com/assets/501861/8614827/3874baf6-2720-11e5-8b89-9ead127a95e3.png)

**Comment by mholt** [comment]
Ah. Try setting the host to elvizlai....xyz instead of localhost in your Caddyfile. Then you can use `bind 127.0.0.1` to force the virtualhost to only be accessible to localhost. Let me know if that works.

**Comment by ElvizLai** [comment]
I'll try it tomorrow, and give you feedback.

**Comment by ElvizLai** [comment]
@mholt This works!
```
elvizlai.XXXXX.xyz:8000 {
	bind 127.0.0.1
	gzip
	log access.log
	browse /download
}
```

**Comment by mholt** [comment]
Excellent!

**Comment by dpastoor** [comment]
commenting on this issue as it took a while before I looked to this issue (which solved my issue).

The problem what was present for me was using a simple "macroservice" SPA that hits a couple endpoints, I was using caddy to serve the html/js via root, then the api was hitting a R analytics api, and a golang based websocket connection. This all worked great when running locally. When I then fired up ngrok, got the exact same issue as discussed above (404).

```
localhost:3000
root /Users/devin/Repos/testts/build
proxy /api/v1 localhost:8000
proxy /socket.io localhost:5000 {
    websocket
}
```

Simply changing to this fixed it!

```diff
+:3000
-localhost:3000
root /Users/devin/Repos/testts/build
proxy /api/v1 localhost:8000
proxy /socket.io localhost:5000 {
    websocket
}
```
